src/http_parser.cpp

```diff
@@ -146,8 +146,9 @@ namespace libtorrent {
 		// TODO: remove to_string() if we're in C++14
 		auto const i = m_header.find(key.to_string());
 		if (i == m_header.end()) return boost::none;
-		auto const val = std::atol(i->second.c_str());
-		if (val <= 0) return boost::none;
+        char *str;
+		auto const val = std::strtol(i->second.c_str(), &str, 10);
+		if (val <= 0 || *str) return boost::none;
```

Comment:
you're not handling the case where the converted value won't fit in the returned variable. `errno` is set to `ERANGE` and the return value is one of  `LONG_MAX`, `LONG_MIN`, `LLONG_MAX` or `LLONG_MIN`